GoogleForCreators/web-stories-wp
# Fonts: Add logic for getting proper font variants

## Summary

Updates the logic of font variants to be up to date with the inline formatting logic.


## Relevant Technical Choices

- Creates a new util file for RichText for getting an array of unique font variants for the HTML.


## To-do




## User-facing changes

The fonts variants are now loaded correctly which should remove the differences between the editor vs preview vs the story in different browsers.


## Testing Instructions
I
1. Create a text element with text "Lorem ipsum dolor amet".
2. Set the font family to "Roboto" and font weight to bold.
3. Reduce the width of the text box to the smallest width that preserves the current wrapping.
4. Publish the story.
5. View the story on Chrome and Safari
6. Verify that the line breaks are in the same place

II
1. Add 2 Pages
2. Add some text to both pages with Open Sans Condensed.
3. Assign Light on one page as the font weight and bold on the other page, for the same font family
4. Verify that the text looks the same in the editor and in the front-end.


---



Fixes #2234
Fixes #1900
